filesmanager/tests/test_tasks.py

```diff
@@ -0,0 +1,234 @@
+            })
+
+    def test_xblock_download_status_not_found(self):
+        data = {
+            "task_id": 'task_id',
+        }
+
+        request = Mock()
+        request.method = 'POST'
+        request.body = json.dumps(data).encode('utf-8')
+
+        with patch('filesmanager.tasks.create_zip_file_task.AsyncResult') as AsyncResult:
+            AsyncResult.return_value = None
+
+            request = self.xblock.download_status(request)
+
+            self.assertEqual(request.status, '200 OK')  # pylint: disable=no-member
+            body = json.loads(request.body)  # pylint: disable=no-member
+            self.assertDictEqual(body, {
+                'status': 'ERROR',
+                'message': 'Task not found',
+            })
+
+    def test_xblock_download_status_with_exception(self):
+        data = {
+            "task_id": 'task_id',
+        }
+
+        request = Mock()
+        request.method = 'POST'
+        request.body = json.dumps(data).encode('utf-8')
+
+        with patch('filesmanager.tasks.create_zip_file_task.AsyncResult') as AsyncResult:
+            AsyncResult.return_value.status = 'FAILURE'
+            AsyncResult.return_value.result = Exception('Some error')
+
+            request = self.xblock.download_status(request)
+
+            self.assertEqual(request.status, '200 OK')  # pylint: disable=no-member
+            body = json.loads(request.body)  # pylint: disable=no-member
+            self.assertDictEqual(body, {
+                'status': 'ERROR',
+                'result': 'Something went wrong. Please try again the download.',
+            })
+
+    def test_xblock_download_status_without_task_id(self):
+        data = {}
+
+        request = Mock()
+        request.method = 'POST'
+        request.body = json.dumps(data).encode('utf-8')
+
+        request = self.xblock.download_status(request)
+
+        self.assertEqual(request.status, '200 OK')  # pylint: disable=no-member
+        body = json.loads(request.body)  # pylint: disable=no-member
+        self.assertDictEqual(body, {
+            'status': 'ERROR',
+            'message': 'Provide a task ID',
+        })
+
+    @patch('filesmanager.tasks.AssetKey.from_string')
+    @patch('filesmanager.tasks.contentstore')
+    def test_create_zip_file(self, mock_contentstore, mock_from_string):
+        """
+        Test that the create_zip_file_task function creates a zip file with the correct structure.
+        """
+        find_mock = Mock()
+        mock_contentstore.return_value.find = find_mock
+        file_content = FileContent(b'dummy', 5)
+        find_mock.return_value = file_content
+        mock_from_string.return_value = 'asset_key'
+
+        data = {
+            "contents": [self.xblock.directories["children"][1]],
+        }
+
+        result = create_zip_file_task(contents=data["contents"])  # pylint: disable=no-value-for-parameter
+
+        self.assertEqual(result, f"/{DOWNLOADS_FOLDER}/{None}.zip")
```

Comment:
why `None`?

Reply:
A celery task has a notion of self, but when calling it as a function self.request is None.

Reply:
I still don't get why `result` includes a None value, can't we create a mock that's closer to what happens?

Reply:
Updated based on this answer: https://stackoverflow.com/questions/47000371/mocking-celery-self-request-attribute-for-bound-tasks-when-called-directly